src/psyclone/domain/nemo/transformations/nemo_arrayrange2loop_trans.py

```diff
@@ -354,8 +403,29 @@ def validate(self, node, options=None):
             except KeyError:
                 # Variable is not defined
                 pass
+
+            # Make sure this array is not already inside a loop with the
+            # loop_variable_name that we choose. If it is we ignore it (as
+            # usually we will already have some explicit loops above this
+            # construct to apply further parallelisation/optimisations).
+            current = node.parent
+            while not isinstance(current, Routine):
+                if isinstance(current, Loop):
+                    varname = current.variable.name
+                    if varname.lower() == loop_variable_name.lower():
+                        raise TransformationError(
+                            "The config file specifies '{0}' as the name of "
+                            "the iteration variable but this is already used "
+                            "by an ancestor loop variable."
+                            "".format(loop_variable_name))
+                current = current.parent
+            # Note that we don't check if the variable name is used in a
+            # context different that the loop if its an integer. We assume
```

Comment:
"...context other than the loop if it's an integer."

Reply:
Done

Reply:
Not quite!

Reply:
It should be now :)